tests/Utilities/testframework.f90

```diff
@@ -0,0 +1,143 @@
+!> \file
```

Comment:
I was deliberately trying to make it read like gtest which has macros EXPECT_EQ, EXPECT_NEAR. At one point I had a macro that was essentially the same as in gtest, however it looked like it would get hit by fortran's line length limit, so I decided it's best to keep it in pure Fortran.